src/DblParser/Desugar.ml

```diff
@@ -751,6 +764,11 @@ and tr_def ?(public=false) (def : Raw.def) =
   | DOpen(pub, path) -> 
     let public = public || pub in
     [ make (DOpen(public, path)) ]
+  | DRec(pub, defs) when List.for_all node_is_rec_data defs ->
```

Comment:
I mean, a comment describing this edge case.

Reply:
Agreed